### PR TITLE
Corruption check is complete

### PR DESCRIPTION
**Pull request by ladislav-zezula**

### CODE REVIEW

File: deps/pelib/CMakeLists.txt [hunk 1]
```diff
@@ -8,8 +8,9 @@ if(CMAKE_CXX_COMPILER)
 endif()
 
 ExternalProject_Add(pelib-project
-	URL https://github.com/avast-tl/pelib/archive/e93eaa7c150f4608a5d02a67f5edc9e54456fe24.zip
-	URL_HASH SHA256=2ffd7e89451c980a1af6d24d4f6dfbb69a660b06ad5de44c481f6431e21de394
+	URL https://github.com/avast-tl/pelib/archive/19f9a34c0dd10657303e15f7a9d03de9d177e6be.zip
+#	URL https://github.com/avast-tl/pelib/archive/e93eaa7c150f4608a5d02a67f5edc9e54456fe24.zip
```
Comment (metthal) [inline deps/pelib/CMakeLists.txt]: Please, remove these lines which are commented out. Also please add `URL_HASH` with the hash of the new archive so CMake can check whether it downloaded correct archive.

File: src/fileinfo/file_information/file_information.h [hunk 2]
```diff
@@ -37,7 +37,7 @@ class FileInformation
 		std::string endianness;                        ///< endianness
 		std::string manifest;                          ///< XML manifest
 		std::string compactManifest;                   ///< compact version of XML manifest
-		FileHeader header;                             ///< file header
+	    FileHeader header;                             ///< file header
```
Comment (metthal) [inline src/fileinfo/file_information/file_information.h]: Spaces.

File: src/fileinfo/file_information/file_information.h [hunk 3]
```diff
@@ -400,7 +400,8 @@ class FileInformation
 		std::string getNumberOfLoadedSegmentsStr(std::ios_base &(* format)(std::ios_base &)) const;
 		const LoadedSegment& getLoadedSegment(std::size_t index) const;
 		const std::string& getLoaderStatusMessage() const;
-		/// @}
+	    const retdec::fileformat::LoaderErrorInfo & getLoaderErrorInfo() const;
```
Comment (metthal) [inline src/fileinfo/file_information/file_information.h]: Spaces.

File: src/fileinfo/file_information/file_information.h [hunk 4]
```diff
@@ -492,6 +493,7 @@ class FileInformation
 		void setSignatureVerified(bool verified);
 		void setLoadedBaseAddress(unsigned long long baseAddress);
 		void setLoaderStatusMessage(const std::string& statusMessage);
+	    void setLoaderErrorInfo(const retdec::fileformat::LoaderErrorInfo & ldrErrInfo);
```
Comment (metthal) [inline src/fileinfo/file_information/file_information.h]: Spaces.

File: src/fileinfo/file_information/file_information_types/loader_info.h [hunk 5]
```diff
@@ -37,7 +37,9 @@ class LoaderInfo
 		unsigned long long _baseAddress;
 		std::vector<LoadedSegment> _loadedSegments;
 		std::string _statusMessage;
-	public:
+		retdec::fileformat::LoaderErrorInfo _ldrErrInfo;
+
+public:
```
Comment (metthal) [inline src/fileinfo/file_information/file_information_types/loader_info.h]: Missing 1 TAB.

**Comment by ladislav-zezula** [comment]
Fixed in commit 440dddeba61f4b5a11490845001573d1ffd7c936.

**Comment by ladislav-zezula** [comment]
With commit https://github.com/avast-tl/retdec/commit/ba7787d4ded107aa4409fa9adc69da1e4060cf77, I also fixed problems related to delay import loading and parsing.

File: deps/pelib/CMakeLists.txt [hunk 6]
```diff
@@ -8,8 +8,7 @@ if(CMAKE_CXX_COMPILER)
 endif()
 
 ExternalProject_Add(pelib-project
-	URL https://github.com/avast-tl/pelib/archive/e93eaa7c150f4608a5d02a67f5edc9e54456fe24.zip
-	URL_HASH SHA256=2ffd7e89451c980a1af6d24d4f6dfbb69a660b06ad5de44c481f6431e21de394
+	URL https://github.com/avast-tl/pelib/archive/master.zip
```
Comment (metthal) [inline deps/pelib/CMakeLists.txt]: Please do not use branches as reference points. Branches can change over time. Use only commit hashes or tags. Also please add missing `URL_HASH`.

File: include/retdec/fileformat/file_format/file_format.h [hunk 7]
```diff
@@ -21,6 +21,20 @@
...
+/**
+* LoaderErrorInfo - common structure that contains loader error code, error message and user-friendly error message
+*/
+
+struct LoaderErrorInfo
+{
+	LoaderErrorInfo() : loaderErrorCode(0), loaderError(nullptr), loaderErrorUserFriendly(nullptr)
+	{}
+
+	std::uint32_t loaderErrorCode;				// Loader error code, cast to uint32_t
```
Comment (metthal) [inline include/retdec/fileformat/file_format/file_format.h]: There are tabs in-between name of the attribute and comment. Please use spaces here. We use tabs only for indentation.

File: include/retdec/fileformat/file_format/file_format.h [hunk 8]
```diff
@@ -67,6 +81,7 @@ class FileFormat : public retdec::utils::ByteValueStorage, private retdec::utils
 		PdbInfo *pdbInfo;                                                 ///< information about related PDB debug file
 		CertificateTable *certificateTable;                               ///< table of certificates
 		Format fileFormat;                                                ///< format of input file
+		LoaderErrorInfo _ldrErrInfo;									  ///< loader error (e.g. Windows loader error for PE files)
```
Comment (metthal) [inline include/retdec/fileformat/file_format/file_format.h]: There are tabs in-between name of the attribute and comment. Please use spaces here. We use tabs only for indentation.

File: deps/pelib/CMakeLists.txt (hunk 6 above)
Comment (ladislav-zezula) [reply deps/pelib/CMakeLists.txt]: Fixed in commit cab945158749486bd6a92ced84237678963a1291

File: include/retdec/fileformat/file_format/file_format.h (hunk 7 above)
Comment (ladislav-zezula) [reply include/retdec/fileformat/file_format/file_format.h]: Fixed in commit cab945158749486bd6a92ced84237678963a1291

File: include/retdec/fileformat/file_format/file_format.h (hunk 8 above)
Comment (ladislav-zezula) [reply include/retdec/fileformat/file_format/file_format.h]: Fixed in commit cab945158749486bd6a92ced84237678963a1291

**Pull request merged**